crates/sui-graphql-rpc/src/server/sui_sdk_data_provider.rs

```diff
@@ -175,6 +181,65 @@ impl DataProvider for SuiClient {
         Ok(connection)
     }
 
+    async fn fetch_checkpoint_connection(
+        &self,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+    ) -> Result<Connection<String, Checkpoint>> {
+        ensure_forward_pagination(&first, &after, &last, &before)?;
```

Comment:
Don't need to implement this in this PR, but checkpoints are probably a thing that we want to support backward pagination for (the presence of the `descending_order` flag in the existing Read API is an indication of that) -- can we add a TODO for it?